cnd/src/http_api/routes/rfc003/handlers/post_swap.rs

```diff
@@ -1,40 +1,89 @@
 use crate::{
-    connector::Connect,
     http_api::{self, asset::HttpAsset, ledger::HttpLedger, problem},
-    network::DialInformation,
+    network::{DialInformation, SendRequest},
+    seed::{Seed, SwapSeed},
     swap_protocols::{
         asset::Asset,
         ledger::{Bitcoin, Ethereum},
-        rfc003::{self, messages::ToRequest, Ledger, SecretSource},
-        HashFunction, SwapId, Timestamp,
+        metadata_store::{self, Metadata, MetadataStore},
+        rfc003::{
+            self,
+            alice::{self, State},
+            create_ledger_events::CreateLedgerEvents,
+            messages::ToRequest,
+            state_store::StateStore,
+            Ledger, Request, SecretSource, Spawn,
+        },
+        HashFunction, LedgerConnectors, Role, SwapId, Timestamp,
     },
 };
 use bitcoin::Amount as BitcoinAmount;
 use ethereum_support::{Erc20Token, EtherQuantity};
+use futures::Stream;
+use futures_core::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+};
 use http_api_problem::{HttpApiProblem, StatusCode as HttpStatusCode};
 use serde::{Deserialize, Serialize};
 
-pub fn handle_post_swap<C: Connect>(
-    con: C,
+pub fn handle_post_swap<D: Clone + StateStore + MetadataStore + SendRequest + Spawn + SwapSeed>(
+    dependencies: D,
     request_body_kind: SwapRequestBodyKind,
 ) -> Result<SwapCreated, HttpApiProblem> {
     let id = SwapId::default();
 
     match request_body_kind {
         SwapRequestBodyKind::BitcoinEthereumBitcoinErc20Token(body) => {
-            con.initiate_request(id, body.peer.clone(), Box::new(body.clone()))?;
+            let seed = dependencies.swap_seed(id);
```

Comment:
I feel like all of this code could go into a single helper-function `handle_supported_swap` (because it handles all combinations of supported swap requests). The only two parameters should be `dependencies` and `body`.

Reply:
Yes, I had made zero attempt to refactor this code - well spotted.